ddtrace/contrib/starlette/patch.py

```diff
@@ -29,6 +30,27 @@
 )
 
 
+# TODO: Remove in 2.x since this is no longer used
```

Comment:
they are not used internally anymore but the behaviour will not change right? It wont break our existing logic I mean?

Here is the code we have

```
def span_modifier(span, scope):
    resource = get_resource(scope)
    if resource:
        span.resource = "{} {}".format(scope["method"], resource)
```

Reply:
How do you configure the `span_modifier`, are you using the `ddtrace.contrib.asgi.TraceMiddleware` directly?

e.g.

```python
from ddtrace.contrib.asgi import TraceMiddleware
from ddtrace.contrib.starlette.patch import get_resource


def span_modifier(span, scope):
    resource = get_resource(scope)
    if resource:
        span.resource = "{} {}".format(scope["method"], resource)


app = TraceMiddleware(app, span_modifier=span_modifier)
```

are you using Starlette/FastAPI? Is there a reason you are using a span modifier + `get_resource` that is not covered by our out of the box integrations?

Reply:
the span modifier is called at the beginning of the ASGI request, but the new behavior for Starlette (and FastAPI) is to try and update the resource name after every routing handler is called.

... which is why I asked for some more details in how you are using the span modifier... how it impacts you depends on whether you are using our Starlette/FastAPI integration code or not.

Reply:
And you are using Starlette/FastAPI and the OOTB Starlette/FastAPI tracing is not working for you?

```python
from ddtrace import patch_all
patch_all()

from starlette.applications import Starlette

async def get_user(request):
    return JSONResponse({ ... })


app = Starlette(routes=[
    Route('/user/{user_id}', get_user),
])
```

For Starlette (similar for FastAPI) the above should work as expected without needing the `TraceMiddleware`. If there is something about your setup that isn't working for you I'd love to solve that.

Reply:
We do use FastAPI but we wrapped the datadog logging and datadog tracing in 2 libraries to remove logic duplication we had across our backed services. So we ended up create our own TracingMiddleware and LoggingMiddleware. For the TracingMiddleware, we simple inherited it from the OOTB one and changed the span_modifier so that we do not have to think about doing that for every routes in every backend that we have.

according to what you are saying if I have that route for instance:
```python
@router.get("/{user_id}", response_model=ExistingUser)
async def get_user(user_id: str):
...
```

The OOTB Tracing middleware should name the span as `/{user_id}` and not `/123456` ?

Reply:
> The OOTB Tracing middleware should name the span as /{user_id} and not /123456 ?

if you use `patch_all()`/`patch(fastapi=True)`/`ddtrace-run`, yes it will do that. we always try to use the route pattern instead of the full path whenever possible... in fact we'll usually default to `<METHOD> <STATUS CODE>` (e.g. `GET 200`) if we know we cannot get the pattern (general strategy for all of our web framework integrations).

if you don't want to use `patch_all`/`patch`/`ddtrace-run` and instead use `TraceMiddleware` manually only, then there should be no change in behavior in the code that you have. only if we are allowed to patch `starlette`/`fastapi` will there be a change in behavior.


if you did both, I would expect you to end up with 2 spans for the request, one of which will might (unsure/untested, I am guessing) might not get it's resource/tags set properly on it.

Reply:
we didn't do the patch because it was not available at the time and when we updated to 1.x version, we kinda took a shortcut to not have to review how we were using our custom middleware everywhere in all our backends.

So I guess that the revert will be good for us but we will need to get rid of our implementation and use the OOTB one before v2 :)

Reply:
👍🏻 sounds good.

yes, given what you've told me. this PR should be good enough for your to keep your behavior, then I'd encourage you to use the patching for FastAPI since I think it is a better experience anyways (IMHO), and if you have any problems there please open an issue!

I'll work on getting this change out. since it is close to EOD... and the releases take about 4-ish hours, 1.2.1 with this fix might not be available until tomorrow morning/noon EST.